src/result/error.rs

```diff
@@ -10,14 +10,17 @@ pub struct Error<Data: Debug = ()> {
 }
 
 impl<Data: Debug> Error<Data> {
+    /// Create an `Error`.
     pub fn new(status: Status, data: Data) -> Self {
         Self { status, data }
     }
 
+    /// Get error `Status`.
     pub fn status(&self) -> Status {
         self.status
     }
 
+    /// Get error payload.
```

Comment:
I'm not sure if the context used elsewhere is "payload" or "error data". Either works, but we should be consistent.

Reply:
Only one other use of `payload` in this file, so switched both to `data`.